### ISSUE TITLE
"Breakpoint split view demo" link on "JBrowse 2 demos" page is currently broken

### ISSUE THREAD

**Issue by ihh** [GMOD/jbrowse-components#1947]
Navigating to the following link (which is on the demos page):
https://jbrowse.org/code/jb2/v1.2.0/?config=test_data/breakpoint/config.json

I get the following "Fatal error":

~~~~
Error: unable to find ref for undefined
    in s
    in s
    in svg
    in div
    in div
    in div
    in s
    in Suspense
    in div
    in ForwardRef
    in ForwardRef
    in div
    in ForwardRef
    in ForwardRef
    in s
    in s
    in div
    in div
    in div
    in s
    in Unknown
    in s
    in s
    in Al
    in m
    in C
    in t
    in Unknown
    in Suspense
~~~~    

I get this bug on OS X with
- Chrome Version 90.0.4430.93 (Official Build) (x86_64)
- Firefox 87.0 (64 bit)
- Firefox 88.0.1. (64 bit)

**Comment by cmdcolin** [comment]
Thanks for catching this...there were a number of things that sort of contributed to this demo getting broken, but mostly the fact that our integration test was commented out (by me, woops)



See https://github.com/GMOD/jbrowse-components/pull/1951 for a fix and a quick analysis of the contributions to the breakage

**Comment by cmdcolin** [comment]
And to be clear we do have integration tests, they are run using react-testing-library aka @testing-library/react and are run in a fake browser on the command line. They have served us pretty well IMO. If needed, we could make actual browser based tests but we can see that the issue we saw here was really the issue was an issue of the breakpoint split view's existing integration test being commented out

**Comment by cmdcolin** [comment]
And also, to be clear, basically just the demo got broken, the actual app version would work fine.

The issue here is basically a misconfiguration of our demo and the integration test that runs over the demo

**Comment by ihh** [comment]
Thanks for fixing it (I assume it is fixed by #1951 though obviously it is not yet merged into main & so is still broken on the demo page. We can close this issue when it's fixed on the demo page.)

Thanks also for explaining about the tests. So, having tests that were accidentally commented out is obviously a step up from not having tests at all; but allowing commented-out tests to break prod (and our demo page really is our version of "prod") clearly indicates some kind of a process flaw, somewhere. What steps can we take to avoid that in future... maybe a question for @rbuels as well

**Comment by cmdcolin** [comment]
I went ahead and merged a quick fix to main branch so that our demo page works ...try not to let a broken thing lie :)

It is a minimal change to help out the live demo page...we can consider some of the changes in #1951 separately (the rest of the code changes there are just refactorings now)

**Comment by cmdcolin** [comment]
properly fixed in v1.3.0 :)
